interfaces/builtin/basedeclaration.go

```diff
@@ -495,6 +495,7 @@ slots:
     allow-installation:
       slot-snap-type:
         - core
+        - app
```

Comment:
Please add:
```
deny-connection:                                                            
  on-classic: false
```
This makes it so a snap declaration is needed when on all snaps (ie, where upower isn't installed as a deb). We do something similar with network-manager, for example (but don't add anything for auto-connection for upower).